src/logger/assertions.hpp

```diff
@@ -375,15 +375,7 @@ extern void __print_back_trace();
 
 
 #define DASSERT_MSG(condition, fmt, ...)                                \
-  do {                                                                  \
-    if (__builtin_expect(!(condition), 0)) {                            \
-      logstream(LOG_ERROR)                                              \
-        << "Check failed: " << #condition << ":\n";                     \
-      logger(LOG_ERROR, fmt, ##__VA_ARGS__);                            \
-      __print_back_trace();                                             \
-      TURI_LOGGER_FAIL_METHOD("assertion failure");                    \
-    }                                                                   \
-  } while(0)
+  ASSERT_MSG(condition, fmt, ##__VA_ARGS__)
```

Comment:
Worth commenting why these macros do the same thing? Should DASSERT_MSG depend on NDEBUG?

Reply:
`DASSERT_MSG` doesn't depend on `NDEBUG`, as `ASSERT_MSG` is always present. In `NDEBUG`, `DASSERT_MSG` expands to nothing; otherwise, it shares the same logic as `ASSERT_MSG`.

Reply:
@nickjong It does; this is in a large ifdef block.